### ISSUE TITLE
Delete all button not working

### ISSUE THREAD

**Issue by nacesprin** [maildev/maildev#465]
When trying to delete all emails with the trash button, it appears as red for some seconds, but none email are deleted
![image](https://github.com/maildev/maildev/assets/295375/083f68df-d7e7-4879-82d8-d1f70c6ed963)

**Comment by nacesprin** [comment]
Solved: I didn't know I must click twice in order to delete all mails

**Comment by tilleps** [comment]
Upgrading from an older version of Maildev, it was not obvious to me either.  My initial impression was that the red icon indicated a problem / error with deleting all emails.

**Comment by andrelec1** [comment]
> Solved: I didn't know I must click twice in order to delete all mails

save my day...

Think a little popover asking to click again is a good idea.

**Comment by haleybe** [comment]
click twice....

**Comment by mahula** [comment]
> I think is a good idea because you don't remove all emails from accident. But is not intuitive you must click twice.

Yes, but wouldn't a dialog to assure to want to delete all emails be more straight forward and UX-friendlier, since thi s seems to be the way in most contemporary applications?

**Comment by djfarrelly** [comment]
This is resolved in the current UI. The old two-click "arm the red button" pattern that everyone found confusing is gone — the delete-all (trash) button is now a single click that pops a confirmation dialog:

```ts
// packages/ui/src/components/layout/Header.tsx
const handleDeleteAll = () => {
  if (window.confirm('Are you sure you want to delete all emails?')) {
    deleteAllMutation.mutate()
  }
}
```

So one click → "Are you sure you want to delete all emails?" → confirm to delete. That's exactly the clearer confirmation UX suggested in this thread. Closing as resolved.
